core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -388,7 +388,7 @@ class Partition(val topic: String,
     val leaderHWIncremented = inWriteLock(leaderIsrUpdateLock) {
       leaderReplicaIfLocal match {
         case Some(leaderReplica) =>
-          val outOfSyncReplicas = getOutOfSyncReplicas(leaderReplica, replicaMaxLagTimeMs)
+          val outOfSyncReplicas = getOutOfSyncReplicas(leaderReplica, replicaMaxLagTimeMs).filter(r => !r.ignoreIsrCheck)
```

Comment:
@edoardocomar : Got it. Could you try something like the following (no need to initialize _lastCaughtUpTimeMs in Replica)?

diff --git a/core/src/main/scala/kafka/cluster/Partition.scala b/core/src/main/scala/kafka/cluster/Partition.scala
index 46815e1..b80fb5c 100755
--- a/core/src/main/scala/kafka/cluster/Partition.scala
+++ b/core/src/main/scala/kafka/cluster/Partition.scala
@@ -190,7 +190,8 @@ class Partition(val topic: String,
         // construct the high watermark metadata for the new leader replica
         leaderReplica.convertHWToLocalOffsetMetadata()
         // reset log end offset for remote replicas
-        assignedReplicas.filter(_.brokerId != localBrokerId).foreach(_.updateLogReadResult(LogReadResult.UnknownLogReadResult))
+        val unknownLogReadResultWithCurTime = LogReadResult.UnknownLogReadResult.copy(fetchTimeMs = curTimeMs)
+        assignedReplicas.filter(_.brokerId != localBrokerId).foreach(_.updateLogReadResult(unknownLogReadResultWithCurTime))
       }
       (maybeIncrementLeaderHW(leaderReplica), isNewLeader)
     }